system-x/services/cert-manager/src/main/java/software/tnb/certmanager/validation/CertManagerValidation.java

```diff
@@ -0,0 +1,156 @@
+        context.put("dnsNameList", dnsNames);
+        context.put("passwordSecretRef", passwordSecretName);
+        StringWriter writer = new StringWriter();
+        template.merge(context, writer);
+        Map<String, Object> spec = new Yaml().load(writer.toString());
+
+        OpenshiftClient.get().genericKubernetesResources(CERTIFICATE_CTX)
+            .inNamespace(OpenshiftClient.get().getNamespace())
+            .resource(new GenericKubernetesResourceBuilder()
+                .withKind(CERTIFICATE_CTX.getKind())
+                .withNewMetadata()
+                .withName(name)
+                .endMetadata()
+                .withAdditionalProperties(spec)
+                .build()
+            ).create();
+    }
+
+    /**
+     * Creates a service account enabled to read secrets in the current namespace.
+     * A new service account, role and role binding will be created
+     * @param serviceAccount String, the name of the service account
+     */
+    public void createSecretViewer(String serviceAccount) {
+        final String roleName = "secret-viewer";
+        final String roleBingingName = "sa-secret-viewer";
+
+        ServiceAccount sa = new ServiceAccountBuilder()
+            .withNewMetadata().withName(serviceAccount).endMetadata()
+            .withAutomountServiceAccountToken(false)
+            .build();
+        OpenshiftClient.get().serviceAccounts().inNamespace(OpenshiftClient.get().getNamespace()).resource(sa).create();
+
+        List<PolicyRule> policyRuleList = new ArrayList<>();
+        PolicyRule endpoints = new PolicyRule();
+        endpoints.setApiGroups(List.of(""));
+        endpoints.setResources(List.of("secrets"));
+        endpoints.setVerbs(Arrays.asList("get", "list", "watch"));
+        policyRuleList.add(endpoints);
+        Role roleCreated = new RoleBuilder()
+            .withNewMetadata().withName(roleName).withNamespace(OpenshiftClient.get().getNamespace()).endMetadata()
+            .addAllToRules(policyRuleList)
+            .build();
+        OpenshiftClient.get().rbac().roles().resource(roleCreated).create();
+
+        List<Subject> subjects = new ArrayList<>();
+        Subject subject = new Subject();
+        subject.setKind("ServiceAccount");
+        subject.setName(sa.getMetadata().getName());
+        subject.setNamespace(OpenshiftClient.get().getNamespace());
+        subjects.add(subject);
+        RoleRef roleRef = new RoleRef();
+        roleRef.setApiGroup("rbac.authorization.k8s.io");
+        roleRef.setKind("Role");
+        roleRef.setName(roleCreated.getMetadata().getName());
+        RoleBinding roleBindingCreated = new RoleBindingBuilder()
+            .withNewMetadata().withName(roleBingingName).withNamespace(OpenshiftClient.get().getNamespace()).endMetadata()
+            .withRoleRef(roleRef)
+            .addAllToSubjects(subjects)
+            .build();
+        OpenshiftClient.get().rbac().roleBindings().resource(roleBindingCreated).create();
+    }
```

Comment:
the `.create()` calls for role and rolebinding will fail if more than 1 test class in 1 run calls `createSecretViewer` method, as the resources are not removed / not created with `serverSideApply()`

Reply:
changed to `serverSideApply` in https://github.com/tnb-software/TNB/pull/1291